src/python/pants/engine/legacy/graph.py

```diff
@@ -296,6 +389,58 @@ class HydratedTargets(Collection.of(HydratedTarget)):
   """An intransitive set of HydratedTarget objects."""
 
 
+class OwnersRequest(datatype([
+  ('sources', tuple),
+  ('include_dependees', str),
```

Comment:
Can we make this an enum-like thing, or override its `__new__` to only allow three values (`none`, `direct`, `transitive` or something)?

Reply:
I just looked into doing this, and while it's definitely easier to do now, given that these values come from a global option, it would currently cause a fair amount of churn to enumify it without making things redundant. Leaving for now.